controller/src/main/java/io/pravega/controller/eventProcessor/impl/EventProcessorCell.java

```diff
@@ -230,9 +232,12 @@ void store(Position position) {
                        final String readerId,
                        final int index,
                        final CheckpointStore checkpointStore) {
-        this.reader = reader;
-        this.selfWriter = selfWriter;
-        this.checkpointStore = checkpointStore;
+        Preconditions.checkNotNull(eventProcessorConfig);
```

Comment:
This class and it's constructor are package local. Would it be possible to move precondition checks to the publicly accessible methods in this package that call this constructor?